routes/harvest.js

```diff
@@ -54,73 +57,74 @@ function buildFilter(facets) {
+router.get('/:type/:provider/:namespace/:name/:revision', asyncMiddleware(getAll))
+
+async function getAll(request, response) {
+  const coordinates = utils.toEntityCoordinatesFromRequest(request)
+  switch ((request.query.form || 'summary').toLowerCase()) {
+    case 'streamed':
+    case 'raw': {
+      const result = await harvestStore.getAll(coordinates)
+      return response.status(200).send(result)
+    }
+    case 'summary': {
+      const raw = await harvestStore.getAll(coordinates)
+      const filter = await getFilter(coordinates)
+      const summarized = await summarizeService.summarizeAll(coordinates, raw, filter)
+      return response.status(200).send(summarized)
+    }
+    case 'list': {
+      const list = await harvestStore.list(coordinates)
+      return response.status(200).send(list)
     }
-  })
-)
+    default:
+      throw new Error(`Invalid request form: ${request.query.form}`)
+  }
+}
 
 // Get a list of the harvested data that we have that matches the url as a prefix
-router.get(
-  '/:type?/:provider?/:namespace?/:name?/:revision?/:tool?',
-  asyncMiddleware(async (request, response) => {
-    const coordinates = utils.toResultCoordinatesFromRequest(request)
-    const result = await harvestStore.list(coordinates)
-    return response.status(200).send(result)
-  })
-)
+router.get('/:type?/:provider?/:namespace?/:name?/:revision?/:tool?', asyncMiddleware(list))
+
+async function list(request, response) {
+  const coordinates = utils.toResultCoordinatesFromRequest(request)
+  const result = await harvestStore.list(coordinates)
+  return response.status(200).send(result)
+}
 
 // post a request to create a resoruce that is the summary of all harvested data available for
 // the components outlined in the POST body
-router.post(
-  '/status',
-  bodyParser.json(),
-  asyncMiddleware(async (request, response) => {
-    const coordinatesList = request.body.map(entry => EntityCoordinates.fromString(entry))
-    const result = await harvestStore.listAll(coordinatesList, 'result')
-    response.status(200).send(result)
-  })
-)
+router.post('/status', bodyParser.json(), asyncMiddleware(status))
+
+async function status(request, response) {
+  const coordinatesList = request.body.map(entry => EntityCoordinates.fromString(entry))
+  const result = await harvestStore.listAll(coordinatesList, 'result')
+  response.status(200).send(result)
+}
 
 // Post a (set of) component to be harvested
-router.post(
-  '/',
-  permissionCheck('harvest'),
-  bodyParser.json(),
-  asyncMiddleware(async (request, response) => {
-    await harvestService.harvest(request.body)
-    response.sendStatus(201)
-  })
-)
+router.post('/', permissionCheck('harvest'), bodyParser.json(), asyncMiddleware(queue))
+
+async function queue(request, response) {
+  const requests = Array.isArray(request.body) ? request.body : [request.body]
+  const isValid = ajv.validate(harvestSchema, requests)
```

Comment:
get the previous pr in first and then resolve conflicts here and use the validator helper utility